File: csharp/Fac.cs
using System;
using System.Numerics;

namespace csharp
{
    class Fac
    {
        static void Main(string[] args)
        {
            BigInteger res = new BigInteger(0);

            for (int i = 1; i <= 3000; i++)
            {
                res += fac(i);
            }
            Console.Write(res);
        }

        static BigInteger fac(int n)
        {
            BigInteger res = new BigInteger(1);

            for (; n > 1; n--)
            {
                res *= n;
            }

            return res;
        }
    }
}

File: java/fac.java
import java.math.BigInteger;

class Fac {
    public static void main(String[] args) {
        BigInteger res = BigInteger.ZERO;

        for(int i = 1; i<=3000;i++) {
            res = res.add(fac(i));
        }

        System.out.print(res);
    }

    private static BigInteger fac(int n) {
        BigInteger res = BigInteger.ONE;

        for(BigInteger big = BigInteger.valueOf(n);
            !big.equals(BigInteger.ONE);
            big = big.subtract(BigInteger.ONE)) {
            res = res.multiply(big);
        }

        return res;
    }
}